server/models/singleSourceOfTruthModel.js

```diff
@@ -2,21 +2,36 @@ const mongoose = require('mongoose');
 
 const { Schema } = mongoose;
 
-const inputParameterSchema = new Schema({
+const singleParameterSchema = new Schema({
   name: String,
-  value: Schema.Types.Mixed,
+  value: String,
   requireUserInput: Boolean,
+  type: String,
+  isRequired: Boolean,
+  infoText: String,
+  index: Number
+});
+
+const parameterObjectSchema = new Schema({
+  ssotId: mongoose.Types.ObjectId,
+  activityId: String,
+  outputVariable: String,
+  rpaParameters: [ singleParameterSchema ],
 });
 
 const instructionSchema = new Schema({
   type: String,
   name: String,
-  predecessorIds: [mongoose.Types.ObjectId],
-  successorIds: [mongoose.Types.ObjectId],
+  predecessorIds: [String],
+  successorIds: [String],
+  id: String
+});
+
+const rpaAttributesObjectSchema = new Schema({
+  ssotId: mongoose.Types.ObjectId,
+  activityId: String,
   rpaApplication: String,
   rpaTask: String,
-  rpaParameters: [inputParameterSchema],
-  outputVariable: Schema.Types.Mixed,
 });
```

Comment:
Why do we have on object for Attributes and one for Parameters and not just one for both? Probably there is a good reason and I would like to understand this

Reply:
We imagined scenarios, where it would be useful, to only retrieve one object or the other.